src/selectors/per-thread/stack-sample.js

```diff
@@ -84,12 +84,6 @@ export function getStackAndSampleSelectorsPerThread(
     }
   );
 
-  const getCallNodeMaxDepth: Selector<number> = createSelector(
-    threadSelectors.getFilteredThread,
```

Comment:
Here's the REAL bug before I did all this other work. This was passing in the thread, NOT the samples-like table. When viewing a native allocations profile, this gives the very wrong result. As long as the stacks in the samples are as deep as the native allocations, this is fine. However, in my dhat importer, the samples have a stack depth of 0, while the native allocations have a large stack depth.

Reply:
However, the other changes I made were related to correctness and being explicit about what is being measured here. I left a few STR in my main comment outlining things that were subtly incorrect.